compiler/rustc_target/src/abi/call/bpf.rs

```diff
@@ -0,0 +1,31 @@
+// see BPFCallingConv.td
+use crate::abi::call::{ArgAbi, FnAbi};
+
+fn classify_ret<Ty>(ret: &mut ArgAbi<'_, Ty>) {
+    if ret.layout.is_aggregate() || ret.layout.size.bits() > 64 {
+        ret.make_indirect();
+    } else {
+        ret.extend_integer_width_to(64);
```

Comment:
AFAIU when alu32 feature is enabled, the 32-bit values are expected to be passed in the `w*` registers, which I think means that we want to conditionally extend to 32 bits here when that feature is enabled.

Perhaps you could always extend to just 32 bits and let LLVM handle extending to 64 bits if alu32 is not enabled?

Reply:
Yes good point, I've done that.